sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetOptions.scala

```diff
@@ -69,12 +69,25 @@ class ParquetOptions(
     .get(MERGE_SCHEMA)
     .map(_.toBoolean)
     .getOrElse(sqlConf.isParquetSchemaMergingEnabled)
+
+  /**
+   * How to resolve duplicated field names. By default, parquet data source fails when hitting
+   * duplicated field names in case-insensitive mode. When converting hive parquet table to parquet
+   * data source, we need to ask parquet data source to pick the first matched field - the same
+   * behavior as hive parquet table - to keep behaviors consistent.
+   */
+  val duplicatedFieldsResolutionMode: String = {
+    parameters.getOrElse(DUPLICATED_FIELDS_RESOLUTION_MODE,
```

Comment:
I don't think we should leave this for Parquet options for now.  Can we just have a SQL config to control this?

Reply:
whether we have a SQL config for it or not, we must define an option here. The conversion happens per-query, so we must have a per-query option to switch the behavior, instead of a per-session SQL config.

Reply:
The conversion itself happens per query but my impression is that the different values don't usually happen in per-query. I mean, I was wondering if users want to set this query by query.

Reply:
I agree this is a little unusual. Usually we have a SQL config first, then we create an option for it if necessary. In this case, we are not adding a config/option from user's requirement, but we need it for an internal optimization.

If we can I would suggest we make it an internal option. But anyway we shouldn't rush to add a SQL config, until we get requirement from users.